### ISSUE TITLE
Class/File naming problem - two files same name, different capitalisation - fails updates on case insensitive filesystems

### ISSUE THREAD

**Issue by czmole** [googleapis/google-api-php-client-services#590]
On running  composer update on a project I got into this problem

```
120/121 [===========================>]  99%    Failed to extract google/apiclient-services: (1) '/usr/bin/unzip' -qq '/Users/...../vendor/composer/tmp-725348a2d9978d55ab03275f7d756ba6' -d '/Users/...../vendor/composer/4f4c8b49'

replace /Users/...../vendor/composer/4f4c8b49/googleapis-google-api-php-client-services-d6bf710/src/ArtifactRegistry/Resource/ProjectsLocationsRepositoriesAptartifacts.php? [y]es, [n]o, [A]ll, [N]one, [r]ename:  NULL
(EOF or read error, treating as "[N]one" ...)

    The archive may contain identical file names with different capitalization (which fails on case insensitive filesystems)
    Unzip with unzip command failed, falling back to ZipArchive class
```

There are two files 

1. /src/ArtifactRegistry/Resource/ProjectsLocationsRepositoriesAptartifacts.php
2. /src/ArtifactRegistry/Resource/ProjectsLocationsRepositoriesAptArtifacts.php

diff content resulted from an auto-update https://github.com/googleapis/google-api-php-client-services/pull/589

**Comment by alebedev80** [comment]
same problem on my mac with case-insensitive filesystem

**Comment by erikn69** [comment]
+1

**Comment by PaolaRuby** [comment]
+1

**Comment by johannesnagl** [comment]
+1

**Comment by fredsal** [comment]
Same 
```
Class Google\Service\ArtifactRegistry\Resource\ProjectsLocationsRepositoriesAptartifacts located in 
    /vendor/google/apiclient-services/src\ArtifactRegistry\Resource\ProjectsLocationsRepositoriesAptArtifacts.php
    does not comply with psr-4 autoloading standard. Skipping.
Class Google\Service\ArtifactRegistry\Resource\ProjectsLocationsRepositoriesGoogetartifacts located in 
    /vendor/google/apiclient-services/src\ArtifactRegistry\Resource\ProjectsLocationsRepositoriesGooGetArtifacts.php
    does not comply with psr-4 autoloading standard. Skipping.
Class Google\Service\ArtifactRegistry\Resource\ProjectsLocationsRepositoriesYumartifacts located in 
    /vendor/google/apiclient-services/src\ArtifactRegistry\Resource\ProjectsLocationsRepositoriesYumArtifacts.php
    does not comply with psr-4 autoloading standard. Skipping.

```

**Comment by bshaffer** [comment]
Thank you for reporting this, it has been resolved in #606 and in [v0.233.1](https://github.com/googleapis/google-api-php-client-services/releases/tag/v0.233.1). Sorry for the inconvenience.

I've added #607 to build on MacOS (case-insensitive filesystem) so this won't happen again.
